src/provider.py

```diff
@@ -184,3 +186,24 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
             self.charm.model.get_relation(PEER).data[self.charm.app].update(
                 {provider_relation_config["username"]: ""}
             )
+
+    def update_connection_info(self):
+        """Updates all relations with current uri, endpoints and tls data.
+
+        If information didn't change, no events will trigger.
+        """
+        bootstrap_server = self.charm.kafka_config.bootstrap_server
+        endpoints = [server.split(":")[0] for server in bootstrap_server]
+        zookeeper_uris = self.charm.kafka_config.zookeeper_config.get("connect", "")
+        tls = "enabled" if self.charm.tls.enabled else "disabled"
+
+        relation_config = {
+            "endpoints": ",".join(endpoints),
+            "uris": ",".join(bootstrap_server),
+            "zookeeper-uris": zookeeper_uris,
+            "tls": tls,
+        }
+
+        for relation in self.charm.model.relations[REL_NAME]:
+            if self.charm.app_peer_data.get(f"relation-{relation.id}", None):
```

Comment:
Worth a comment here explaining that you're using this to check for active Client Relations